### PR TITLE
refactor: refactor file transfer service

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link

None

### Change description

This PR does:
- move service into modules directory
- implements consistent error handling
- add support for scenarios where file is either infected or not scanned
- implements consistent unit tests

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are commited to keeping commit history clean, consistent and linear. To achive this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/modules/file-transfer-service/FileTransferService.ts [hunk 1]
```diff
@@ -120,15 +101,22 @@ export class FileTransferService {
...
     }
 
-    private getErrorFrom(err: any, fileId: string): Error {
-
-        if (err.isAxiosError) {
-            switch (err.response.status) {
-                case NOT_FOUND:
-                    return new Error(`File ${fileId} not found.`);
+    private handleResponseError(operation: 'upload' | 'metadata retrieval' | 'download' | 'deletion', subject: string):
+        (err: AxiosError) => never {
```
Comment (n-r-b) [inline src/modules/file-transfer-service/FileTransferService.ts:105]: does this mean that the following (within `{ }`) should 'never' occur?

File: src/modules/file-transfer-service/FileTransferService.ts [hunk 2]
```diff
@@ -120,15 +101,22 @@ export class FileTransferService {
...
+        // tslint:disable: max-line-length
+        return (err: AxiosError) => {
+            if (err.isAxiosError) {
```
Comment (n-r-b) [inline src/modules/file-transfer-service/FileTransferService.ts:108]: wouldn't the `err` always be of type `AxiosError`?

File: src/modules/file-transfer-service/FileTransferService.ts (hunk 1 above)
Comment (geek-soft) [reply src/modules/file-transfer-service/FileTransferService.ts:105]: Never type is required by `catch` function. I started with `void` but `never` was required.

File: src/modules/file-transfer-service/FileTransferService.ts (hunk 2 above)
Comment (geek-soft) [reply src/modules/file-transfer-service/FileTransferService.ts:108]: Hmm, not sure, Francisco started this convention and I think it gives us extra safety.

File: src/modules/file-transfer-service/FileTransferService.ts [hunk 3]
```diff
@@ -120,15 +101,22 @@ export class FileTransferService {
...
+                switch (err.response!.status) {
+                    case FORBIDDEN:
```
Comment (n-r-b) [inline src/modules/file-transfer-service/FileTransferService.ts:110]: To me, 'FORBIDDEN' means unauthorised, not 'in progress'....however i don't really know of a more suitable error code for this

File: src/modules/file-transfer-service/FileTransferService.ts (hunk 2 above)
Comment (geek-soft) [reply src/modules/file-transfer-service/FileTransferService.ts:108]: This is what docs say: https://github.com/axios/axios#handling-errors

It does not explain what errors you can get but with HTTP which is complex topic I feel more safe with that condition. In fact you made me consider another case where response has not been received. I will make it even safer.

Comment (geek-soft) [reply src/modules/file-transfer-service/FileTransferService.ts:108]: Now we will have:

```
            if (err.isAxiosError && err.response != null) {
                switch (err.response.status) {
```

File: src/modules/file-transfer-service/FileTransferService.ts (hunk 3 above)
Comment (geek-soft) [reply src/modules/file-transfer-service/FileTransferService.ts:110]: Well this is not in my hands, this is how API behaves: https://github.com/companieshouse/file-transfer-api/blob/master/src/routes/file-download.ts#L16

I basically went through an API code and reflected all scenarios in the service code.

Comment (n-r-b) [reply src/modules/file-transfer-service/FileTransferService.ts:110]: ok, fair enough 👍

File: src/modules/file-transfer-service/FileTransferService.ts (hunk 2 above)
Comment (n-r-b) [reply src/modules/file-transfer-service/FileTransferService.ts]: i like the extra safety!

**Pull request merged**